packages/react-internal/src/components/SpinButton/SpinButton.base.tsx

```diff
@@ -156,14 +159,20 @@ export const SpinButtonBase: React.FunctionComponent<ISpinButtonProps> = React.f
     if (onValidate) {
       return onValidate(newValue, ev);
     }
-    if (!newValue || newValue.trim().length === 0 || isNaN(Number(newValue))) {
-      return internalState.lastValidValue;
+    // default validation handling
+    let newNumber = Number(newValue);
+    if (newValue && newValue.trim().length && !isNaN(newNumber)) {
+      if (typeof min === 'number') {
+        newNumber = Math.max(min, newNumber);
+      }
+      if (typeof max === 'number') {
+        newNumber = Math.min(max, newNumber);
+      }
+      return String(newNumber);
```

Comment:
So, in this case, would it return `undefined` in scenarios that don't enter that if statement?

Reply:
Yes, I think that's the same behavior as before, and in this context it means the update is invalid and should be ignored.

Reply:
Oh wait actually it's different (bad interaction of this change with another one). I'll fix that.

Reply:
Hmm never mind, while the behavior of that particular function is slightly different, the overall behavior is still the same. So I'm going to leave it as-is.